docs/flow/modin/core/dataframe/base/index.rst

```diff
@@ -1,3 +1,17 @@
+Purpose
+=======
+``BaseDataframe`` serves the purpose of describing and defining the internal dataframe algebra.
+
+It is the core construction element which serves as a middle layer between high-level pandas-imitating API
+and lower-level query compiler. ``BaseDataframe`` is the interface which implementations actually translate the dataframe algebra calls to queries to the underlying compiler.
+
+The purpose of such translation is to reduce the vast amount of public pandas API to something smaller but sufficient.
+Query compiler level could reduce the API even further depending on the implementation.
+
+Base dataframe and axis partitions are the interfaces that must be implemented by any backend that wants to be plugged in Modin.
+These classes are mostly abstract, however very simple and generic enough methods like
```

Comment:
I would make base classes just declare interfaces.

Reply:
again, out of scope of docs PR 😉

Reply:
yep